packages/sitecore-jss-react/src/components/BYOCRenderer.tsx

```diff
@@ -0,0 +1,145 @@
+import React from 'react';
+import { ComponentFields } from '@sitecore-jss/sitecore-jss/layout';
+import { getDataFromFields } from '../utils';
+import { MissingComponent, MissingComponentProps } from './MissingComponent';
+import { RegisteredComponents } from '@sitecore-feaas/clientside/types/ui/FEAASExternal';
+
+/**
+ * Data from rendering params on Sitecore's BYOC rendering
+ */
+export type BYOCRenderingParams = {
+  /**
+   * Name of the component to render
+   */
+  ComponentName: string;
+  /**
+   * JSON props to pass into rendered component
+   */
+  ComponentProps?: string;
+  /**
+   * A string with classes that can be used to apply themes, via SXA functionality
+   */
+  styles?: string;
+  RenderingIdentifier?: string;
+};
+
+/**
+ * Props for BYOC wrapper component
+ */
+export type BYOCProps = {
+  /**
+   * rendering params
+   */
+  params?: BYOCRenderingParams;
+  /**
+   * fields from datasource items to be passed as rendered child component props
+   */
+  fields?: ComponentFields;
+};
+
+/**
+ * Props for BYOCRenderer component. Includes components list to load external components from.
+ */
+type BYOCRendererProps = BYOCProps & {
```

Comment:
We should export this type as well (all the way up through sitecore-jss-nextjs), and add doc comments for the props.